src/quote/flashmint/hyeth/swap-data.ts

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { AddressZero, EthAddress } from 'constants/addresses'
+import { noopSwapData } from 'constants/swapdata'
+import { ETH, USDC, WETH } from 'constants/tokens'
+import { QuoteToken } from 'quote/interfaces'
+import { Exchange, SwapData } from 'utils'
+
+export function getComponentsSwapData(isMinting: boolean): SwapData[] {
+  const path = isMinting
+    ? [WETH.address!, USDC.address!]
+    : [USDC.address!, WETH.address!]
+  return [
+    noopSwapData,
+    noopSwapData,
+    noopSwapData,
+    noopSwapData,
+    noopSwapData,
+    {
+      exchange: Exchange.UniV3,
+      fees: [500],
+      path,
+      pool: AddressZero,
+    },
+  ]
+}
```

Comment:
@ckoopmann please check if the swap data generation looks good to you in all three functions.